packages/replay/src/replay.ts

```diff
@@ -289,6 +291,36 @@ export class ReplayContainer implements ReplayContainerInterface {
     this.startRecording();
   }
 
+  /**
+   * If not in "session" recording mode, flush event buffer (i.e. creates a new replay).
+   * Unless `continueRecording` is false, the replay will continue to record and
+   * behave as a "session"-based replay.
+   */
+  public async sendBufferedReplayOrFlush({ continueRecording = true }: SendBufferedReplayOptions = {}): Promise<void> {
+    // Don't allow if in session mode, use `flush()` instead
+    if (this.recordingMode === 'session') {
+      return this._debouncedFlush() as Promise<void>;
```

Comment:
I think this is preferred, I don't really see a reason they would need to flush immediate in session mode.

Reply:
Hmm good question... But then this is pretty much useless in session mode, right? I think I tend to make this flush immediately (also makes this less "breaking" if somebody has been using this already), as it's also more consistent:

* Calling `flush` will _always_ flush "immediately"
  * If you've been in error mode, it will convert to a session and continue
  * If you've already been in session mode, it will just continue normally

Sounds relatively easy to teach/document to me?

Reply:
Yeah you're right, I suppose I don't like the idea of exposing flush immediate as it can lead to mistakes that end up flooding our backend.

Reply:
I mean worst case it will be rate limited, right? 😅 Also actually changing this would kind of be a breaking change, as it currently will flush immediately?

Reply:
Hah yeah I like to protect the user from themselves, technically it'd be a breaking change but I think the few users that were using flush were probably using it in buffer mode? Regardless, I've updated it to flush immediately.